src/Core/Framework/DataAbstractionLayer/DefinitionInstanceRegistry.php

```diff
@@ -13,33 +13,22 @@
 #[Package('framework')]
 class DefinitionInstanceRegistry
 {
-    /**
-     * @var ContainerInterface
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $container;
+    protected ContainerInterface $container;
```

Comment:
we could use constructor promotion here and for the other two constructor args, then we also don't have to repeat the array shape annotation